awswrangler/s3/_read.py

```diff
@@ -143,3 +144,19 @@ def _read_dfs_from_multiple_paths(
         partial_read_func = partial(read_func, **kwargs)
         versions = [version_ids.get(p) if isinstance(version_ids, dict) else None for p in paths]
         return list(df for df in executor.map(partial_read_func, paths, versions))
+
+
+def _read_tables_from_multiple_paths(
+    read_func: Callable[..., pd.DataFrame],
+    paths: List[str],
+    use_threads: Union[bool, int],
+    kwargs: Dict[str, Any],
+) -> List[Table]:
+    cpus = ensure_cpu_count(use_threads)
+    if cpus < 2:
+        return [read_func(path, **kwargs) for path in paths]
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=ensure_cpu_count(use_threads)) as executor:
+        kwargs["boto3_session"] = boto3_to_primitives(kwargs["boto3_session"])
+        partial_read_func = partial(read_func, **kwargs)
```

Comment:
Is partial application needed here? You can pass other arguments to map using `itertools.repeat` - see example in the current codebase.

Reply:
If you look at the function just above this one, it's doing the exact same thing. The only difference is that the other handles version_id which sadly does not exist in S3 select